Update the auto-generated _get_type() functions so that they use
g_once_init_{enter,leave}_pointer where appropriate.

See the upstream issue:
  https://gitlab.gnome.org/GNOME/glib/-/issues/2842
for more information.

Upstream-Status: Inappropriate [CHERI specific]

--- a/util/cairo-gobject/cairo-gobject-structs.c	2018-08-17 02:10:53.000000000 +0100
+++ b/tuil/cairo-gobject/cairo-gobject-structs.c	2025-02-03 16:34:12.754565238 +0000
@@ -44,12 +44,12 @@
 GType \
 underscore_name ## _get_type (void) \
 { \
-   static volatile gsize type_volatile = 0; \
-   if (g_once_init_enter (&type_volatile)) { \
+   static volatile GType type_volatile = 0; \
+   if (g_once_init_enter_pointer (&type_volatile)) { \
       GType type = g_boxed_type_register_static (g_intern_static_string (Name), \
                                                  (GBoxedCopyFunc)copy_func, \
                                                  (GBoxedFreeFunc)free_func); \
-      g_once_init_leave (&type_volatile, type); \
+      g_once_init_leave_pointer (&type_volatile, type); \
    } \
    return type_volatile; \
 }
--- a/util/cairo-gobject/cairo-gobject-enums.c	2018-08-17 02:10:53.000000000 +0100
+++ b/util/cairo-gobject/cairo-gobject-enums.c	2025-02-03 16:38:02.348820846 +0000
@@ -10,8 +10,8 @@
 GType
 cairo_gobject_status_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_STATUS_SUCCESS, "CAIRO_STATUS_SUCCESS", "success" },
           { CAIRO_STATUS_NO_MEMORY, "CAIRO_STATUS_NO_MEMORY", "no-memory" },
@@ -60,7 +60,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_status_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -68,8 +68,8 @@
 GType
 cairo_gobject_content_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_CONTENT_COLOR, "CAIRO_CONTENT_COLOR", "color" },
           { CAIRO_CONTENT_ALPHA, "CAIRO_CONTENT_ALPHA", "alpha" },
@@ -78,7 +78,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_content_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -86,8 +86,8 @@
 GType
 cairo_gobject_operator_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_OPERATOR_CLEAR, "CAIRO_OPERATOR_CLEAR", "clear" },
           { CAIRO_OPERATOR_SOURCE, "CAIRO_OPERATOR_SOURCE", "source" },
@@ -122,7 +122,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_operator_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -130,8 +130,8 @@
 GType
 cairo_gobject_antialias_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_ANTIALIAS_DEFAULT, "CAIRO_ANTIALIAS_DEFAULT", "default" },
 
@@ -147,7 +147,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_antialias_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -155,8 +155,8 @@
 GType
 cairo_gobject_fill_rule_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_FILL_RULE_WINDING, "CAIRO_FILL_RULE_WINDING", "winding" },
           { CAIRO_FILL_RULE_EVEN_ODD, "CAIRO_FILL_RULE_EVEN_ODD", "even-odd" },
@@ -164,7 +164,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_fill_rule_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -172,8 +172,8 @@
 GType
 cairo_gobject_line_cap_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_LINE_CAP_BUTT, "CAIRO_LINE_CAP_BUTT", "butt" },
           { CAIRO_LINE_CAP_ROUND, "CAIRO_LINE_CAP_ROUND", "round" },
@@ -182,7 +182,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_line_cap_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -190,8 +190,8 @@
 GType
 cairo_gobject_line_join_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_LINE_JOIN_MITER, "CAIRO_LINE_JOIN_MITER", "miter" },
           { CAIRO_LINE_JOIN_ROUND, "CAIRO_LINE_JOIN_ROUND", "round" },
@@ -200,7 +200,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_line_join_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -208,15 +208,15 @@
 GType
 cairo_gobject_text_cluster_flags_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_TEXT_CLUSTER_FLAG_BACKWARD, "CAIRO_TEXT_CLUSTER_FLAG_BACKWARD", "backward" },
           { 0, NULL, NULL }
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_text_cluster_flags_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -224,8 +224,8 @@
 GType
 cairo_gobject_font_slant_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_FONT_SLANT_NORMAL, "CAIRO_FONT_SLANT_NORMAL", "normal" },
           { CAIRO_FONT_SLANT_ITALIC, "CAIRO_FONT_SLANT_ITALIC", "italic" },
@@ -234,7 +234,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_font_slant_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -242,8 +242,8 @@
 GType
 cairo_gobject_font_weight_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_FONT_WEIGHT_NORMAL, "CAIRO_FONT_WEIGHT_NORMAL", "normal" },
           { CAIRO_FONT_WEIGHT_BOLD, "CAIRO_FONT_WEIGHT_BOLD", "bold" },
@@ -251,7 +251,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_font_weight_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -259,8 +259,8 @@
 GType
 cairo_gobject_subpixel_order_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_SUBPIXEL_ORDER_DEFAULT, "CAIRO_SUBPIXEL_ORDER_DEFAULT", "default" },
           { CAIRO_SUBPIXEL_ORDER_RGB, "CAIRO_SUBPIXEL_ORDER_RGB", "rgb" },
@@ -271,7 +271,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_subpixel_order_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -279,8 +279,8 @@
 GType
 cairo_gobject_hint_style_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_HINT_STYLE_DEFAULT, "CAIRO_HINT_STYLE_DEFAULT", "default" },
           { CAIRO_HINT_STYLE_NONE, "CAIRO_HINT_STYLE_NONE", "none" },
@@ -291,7 +291,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_hint_style_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -299,8 +299,8 @@
 GType
 cairo_gobject_hint_metrics_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_HINT_METRICS_DEFAULT, "CAIRO_HINT_METRICS_DEFAULT", "default" },
           { CAIRO_HINT_METRICS_OFF, "CAIRO_HINT_METRICS_OFF", "off" },
@@ -309,7 +309,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_hint_metrics_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -317,8 +317,8 @@
 GType
 cairo_gobject_font_type_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_FONT_TYPE_TOY, "CAIRO_FONT_TYPE_TOY", "toy" },
           { CAIRO_FONT_TYPE_FT, "CAIRO_FONT_TYPE_FT", "ft" },
@@ -329,7 +329,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_font_type_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -337,8 +337,8 @@
 GType
 cairo_gobject_path_data_type_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_PATH_MOVE_TO, "CAIRO_PATH_MOVE_TO", "move-to" },
           { CAIRO_PATH_LINE_TO, "CAIRO_PATH_LINE_TO", "line-to" },
@@ -348,7 +348,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_path_data_type_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -356,8 +356,8 @@
 GType
 cairo_gobject_device_type_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_DEVICE_TYPE_DRM, "CAIRO_DEVICE_TYPE_DRM", "drm" },
           { CAIRO_DEVICE_TYPE_GL, "CAIRO_DEVICE_TYPE_GL", "gl" },
@@ -369,7 +369,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_device_type_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -377,8 +377,8 @@
 GType
 cairo_gobject_surface_type_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_SURFACE_TYPE_IMAGE, "CAIRO_SURFACE_TYPE_IMAGE", "image" },
           { CAIRO_SURFACE_TYPE_PDF, "CAIRO_SURFACE_TYPE_PDF", "pdf" },
@@ -407,7 +407,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_surface_type_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -415,8 +415,8 @@
 GType
 cairo_gobject_format_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_FORMAT_INVALID, "CAIRO_FORMAT_INVALID", "invalid" },
           { CAIRO_FORMAT_ARGB32, "CAIRO_FORMAT_ARGB32", "argb32" },
@@ -428,7 +428,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_format_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -436,8 +436,8 @@
 GType
 cairo_gobject_pattern_type_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_PATTERN_TYPE_SOLID, "CAIRO_PATTERN_TYPE_SOLID", "solid" },
           { CAIRO_PATTERN_TYPE_SURFACE, "CAIRO_PATTERN_TYPE_SURFACE", "surface" },
@@ -448,7 +448,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_pattern_type_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -456,8 +456,8 @@
 GType
 cairo_gobject_extend_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_EXTEND_NONE, "CAIRO_EXTEND_NONE", "none" },
           { CAIRO_EXTEND_REPEAT, "CAIRO_EXTEND_REPEAT", "repeat" },
@@ -467,7 +467,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_extend_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -475,8 +475,8 @@
 GType
 cairo_gobject_filter_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_FILTER_FAST, "CAIRO_FILTER_FAST", "fast" },
           { CAIRO_FILTER_GOOD, "CAIRO_FILTER_GOOD", "good" },
@@ -488,7 +488,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_filter_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
@@ -496,8 +496,8 @@
 GType
 cairo_gobject_region_overlap_get_type (void)
 {
-   static volatile gsize type_volatile = 0;
-   if (g_once_init_enter (&type_volatile)) {
+   static volatile GType type_volatile = 0;
+   if (g_once_init_enter_pointer (&type_volatile)) {
       static const GEnumValue values[] = {
           { CAIRO_REGION_OVERLAP_IN, "CAIRO_REGION_OVERLAP_IN", "in" },
           { CAIRO_REGION_OVERLAP_OUT, "CAIRO_REGION_OVERLAP_OUT", "out" },
@@ -506,7 +506,7 @@
       };
       GType type = g_enum_register_static (g_intern_static_string ("cairo_region_overlap_t"), values);
 
-      g_once_init_leave (&type_volatile, type);
+      g_once_init_leave_pointer (&type_volatile, type);
    }
    return type_volatile;
 }
